x-pack/platform/plugins/shared/timelines/server/search_strategy/timeline/factory/helpers/get_nested_parent_path.ts

```diff
@@ -5,14 +5,10 @@
  * 2.0.
  */
 
-import { Fields } from '../../../../../common/search_strategy';
-
 /**
  * If a prefix of our full field path is present as a field, we know that our field is nested
  */
 export const getNestedParentPath = (
   fieldPath: string,
-  fields: Fields | undefined
-): string | undefined =>
-  fields &&
-  Object.keys(fields).find((field) => field !== fieldPath && fieldPath.startsWith(`${field}.`));
+  fields: string[],
+): string | undefined => fields.find((field) => field !== fieldPath && fieldPath.startsWith(`${field}.`));
```

Comment:
This is the primary change. `getNestedParentPath` is called in a few looping constructs, and hitting `Object.keys()` on each iteration wasn't performant